src/uk/gov/defra/ffc/DefraUtils.groovy

```diff
@@ -183,6 +183,33 @@ def triggerDeploy(jenkinsUrl, jobName, token, params) {
   sh(script: "curl -k $url")
 }
 
+def releaseExists(containerTag, repoName, token){
+    try {
+      def result = sh(returnStdout: true, script: "curl -s -H 'Authorization: token $token' https://api.github.com/repos/DEFRA/$repoName/releases/tags/$containerTag | jq '.tag_name'").trim().replaceAll (/"/, '') == "$containerTag" ? true : false
+      return result
+    }
+    catch(Exception ex) {
+      echo "Failed to check release status on github"
+      throw new Exception (ex)
+    }  
+}
+
+def triggerRelease(containerTag, repoName, releaseDescription, token){
+    if (releaseExists(containerTag, repoName, token)){
+      echo "Release $containerTag already exists"
+      return
+    }
+
+    echo "Triggering release $containerTag for $repoName"
+    boolean result = false
+    result = sh(returnStdout: true, script: "curl -s -X POST -H 'Authorization: token $token' -d '{ \"tag_name\" : \"$containerTag\", \"name\" : \"Release $containerTag\", \"body\" : \" Release $releaseDescription\" }' https://api.github.com/repos/DEFRA/$repoName/releases")
+    echo "The release result is $result"
+
+    if (releaseExists(containerTag, repoName, token)){
+      echo "Release Successful"
+    }
```

Comment:
If the release doesn't work should we throw an exception and cause the pipeline to fail?  Otherwise no one will know it didn't work.

Reply:
This has been added as suggested.